### ISSUE TITLE
Rename unsafe lifecycle methods

### ISSUE THREAD

**Issue by wlthomson** [openmsupply/mobile#1398]
## The suggestion

Unsafe lifecycle methods are getting renamed in React `v16.9.0` (https://reactjs.org/blog/2019/08/08/react-v16.9.0.html):

```
componentWillMount -> UNSAFE_componentWillMount
componentWillReceiveProps -> UNSAFE_componentWillReceiveProps
componentWillUpdate -> UNSAFE_componentWillUpadte
```

The old names will continue to work for now and in `17.x`, but using them will throw warnings. It's probably a good idea to take the official React advice and rename them (and continue the process of migrating completely to hooks).

**Why?**

See above.

## Describe alternatives you've considered

Kicking it down the road until later.

## Additional context

See #1393 for more context.

**Comment by joshxg** [comment]
I think this should be done.

I's recommend the tissue to use https://github.com/reactjs/react-codemod for the easiest PR ever

**Comment by joshxg** [comment]
Fixed by #1438

**Comment by joshxg** [comment]
Internal development change - moving to done
